src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -460,6 +446,18 @@ sealed trait ZPure[-S1, +S2, -R, +E, +A] { self =>
     self.flatMap(a => that.flatMap(b => State.succeed(f(a, b))))
 }
 
+trait ZPureSyntax {
+  final implicit class ZPureOps[S1, S2, R, E, A](private val self: ZPure[S1, S2, R, E, A]) {
+
+    /**
+     * Executes this computation and returns its value, if it succeeds, but
+     * otherwise succeeds with the specified value.
+     */
+    def orElseSucceed[A1 >: A](a1: => A1)(implicit ev: S1 <:< S2): ZPure[S1, S2, R, Nothing, A1] =
```

Comment:
If you are going to require implicit evidence here that `S1 <:< S2`, then you may as well just have one type parameter `S` on the `ZPureOps`. It's the same thing only will infer better.

Reply:
Yes. That is the approach I took with `UnifiedSyntax` in #461.